### PR TITLE
Anticheat: Use the physics_override modifiers aswell

### LINKED ISSUE #4291: Anticheat false positives
If anticheat is on, and physics override is greater than 1 (or whatever movement_speed_walk is), then anticheat resets position constantly.  This only occurs when you lack the fast priv!

https://www.youtube.com/watch?v=_FjERImCPxk

### PR DESCRIPTION
**Pull request by SmallJoker**
Attempt to fix #4291
..and tidy up code.

EDIT: The code compiled fine but I have no idea how to test this properly.

### CODE REVIEW

**Comment by everamzah** [comment]
This works!  I am uploading a video now, for no good reason, but yeah -- this doesn't reset my position when I sprint (physics override) while lacking the 'fast' priv.  Bravo!

**Comment by everamzah** [comment]
I'm not too proud to beg.  Please merge this so we can use sprint mods without having to disable _all_ of anticheat.

**Comment by Fixer-007** [comment]
Yes, with this no need to disable anticheat. :+1:

**Comment by ChaosWormz** [comment]
sadly anti cheat still activates with this patch

**Comment by SmallJoker** [comment]
@ChaosWormz Thank you. I haven't checked the behaviour properly. The fast mode is affected by this physics modifier aswell.

File: src/content_sao.cpp [hunk 1]
```diff
@@ -1324,52 +1324,52 @@ std::string PlayerSAO::getPropertyPacket()
...
+	*/
+
+	float player_max_speed = 0;
+
+	if (m_privs.count("fast") != 0) {
+		// Fast speed
+		player_max_speed = m_player->movement_speed_fast * m_physics_override_speed;
+	} else {
+		// Normal speed
+		player_max_speed = m_player->movement_speed_walk * m_physics_override_speed;
```
Comment (SmallJoker) [inline src/content_sao.cpp]: Note to developers: Added `m_physics_override_speed` in both cases.

**Comment by everamzah** [comment]
This please.

**Comment by Fixer-007** [comment]
So much yes!

**Comment by paramat** [comment]
:+1:

File: src/content_sao.cpp [hunk 2]
```diff
@@ -1324,52 +1324,52 @@ std::string PlayerSAO::getPropertyPacket()
...
+	// Tolerance. The lag pool does this a bit.
+	//player_max_speed *= 2.5;
+
+	v3f diff = (m_player->getPosition() - m_last_good_position);
+	float d_vert = diff.Y;
+	diff.Y = 0;
+	float d_horiz = diff.getLength();
+	float required_time = d_horiz / player_max_speed;
+
+	if (d_vert > 0 && d_vert/player_max_speed > required_time)
```
Comment (paramat) [inline src/content_sao.cpp]: Spaces around '/'.

**Comment by paramat** [comment]
Line comment, otherwise seems good.

**Comment by paramat** [comment]
:+1:

File: src/content_sao.cpp (hunk 1 above)
Comment (kwolekr) [inline src/content_sao.cpp]: Why not just player_max_speed *= m_physics_override_speed; after the if statement then?

**Comment by kwolekr** [comment]
LGTM :+1:

**Comment by kwolekr** [comment]
https://github.com/minetest/minetest/commit/c0cd7aa351fc59a6f699c6f945673f02f02e8c32

**Pull request merged**